src/ejabberd_app.erl

```diff
@@ -72,8 +72,8 @@ start(normal, _Args) ->
     %%ejabberd_debug:fprof_start(),
     start_services(),
     start_modules(),
-    mongoose_metrics:init(),
     ejabberd_listener:start_listeners(),
+    mongoose_metrics:init(),
```

Comment:
Yes, `mongoose_metrics:init()` subscribes all reporters to all metrics which are running at the moment of its start. Since `ejabberd_cowboy` is started as one of the listeners, it would define metrics after the reporters were subscribed, which means that no reporter reported HTTP API metrics.